### ISSUE TITLE
Support for azure firewall with forced tunneling

### ISSUE THREAD

**Issue by CoopCNIT** [terraform-providers/terraform-provider-azurerm#8488]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Adding possibility to deploy Azure Firewall with force tunneling enabled.

### New or Affected Resource(s)

* azurerm_firewall

### Potential Terraform Configuration

```hcl
resource "azurerm_firewall" "firewall" {
  name                = "firewall-test"
  location            = var.location
  resource_group_name = azurerm_resource_group.main.name
  force_tunneling     = enabled

  ip_configuration {
    name                 = "configuration"
    subnet_id            = azurerm_subnet.firewall.id
    public_ip_address_id = azurerm_public_ip.firewallpip.id
  }

   mgmt_ip_configuration {
    name                 = "management"
    subnet_id            = azurerm_subnet.firewallmgmt.id
    public_ip_address_id = azurerm_public_ip.firewallmgmtpip.id
  }
}
```

### References

When deploying a azure firewall with force tunneling, you will need another public ip + a subnet called AzureFirewallManagementSubnet (min /26). 

* https://docs.microsoft.com/en-us/azure/firewall/forced-tunneling

**Comment by magodo** [comment]
@CoopCNIT Thank you for submitting this!

Besides, one thing I'm not sure is that I found after enabling forced tunneling on Portal, the `AzureFirewallSubnet` will have an implicit route table created under the scenes. I need to contact the service team to figure out the details so that we can provide a concrete solution.

**Comment by magodo** [comment]
@CoopCNIT Thank you for submitting this!

Since this is already fixed by #8235 (will be available in v2.28.0), I'm going to close this issue.
